content/backend/graphql-scala/3-handling_arguments.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Handling Arguments
+pageTitle: "Handling arguments which are provided in query"
+description: "In this chapter, at first you will learn how to use arguments."
+question: "Does GraphQL needs HTTP Server?"
+answers: ["Yes. It needs HTTP server.", "Yes, it needs HTTP server but some of features can be used without that", "No, but it strictly recommended to use. Without HTTP layer, GraphQL is losing some of its features.","No, GraphQL is specification is far away from tranportation protocol. You can use HTTP, Websockets, sockets or even use it internally in you application." ]
+correctAnswer: 3
+---
+
+
+### Arguments
+
+Let's assume, we want to fetch links for only one link (giving ID of it) or few links. We will need to manage parsing of arguments.
+
+For example such queries:
+
+
+```graphql
+
+query {
+    link(id: 1){
+    	id
+    	name
+  	}
+  	links(ids: [2,3]){
+      id
+      name
+    }  
+}
+```
+What must we do? Firstly add to DAO functions that gives us link by one or more ID's.
+
+<Instruction>
+
+Open `DAO.scala` file, and add following functions:
+
+```scala
+
+def getLink(id: Int): Future[Option[Link]] = db.run(
+   Links.filter(_.id === id).result.headOption
+ )
+
+ def getLinks(ids: Seq[Int]) = db.run(
+   Links.filter(_.id inSet ids).result
+ )
+```
+
+</Instruction>
+
+Next we have to add fields to the main `Query` object with those functions as resolvers.
+
+<Instruction>
+
+Now open `GraphQLSchema.scala` file, and in codeblock `fields[MyContext, Unit](...)` add two additional fields:
+
+```
+Field("link",//1
+  OptionType(LinkType),//2
+  arguments = List(Argument("id", IntType)),//3
+  resolve = c => c.ctx.dao.getLink(c.arg[Int]("id"))//4
+),
+Field("links",//1
+  ListType(LinkType),//2
+  arguments = List(Argument("ids", ListInputType(IntType))),//3
+  resolve = c => c.ctx.dao.getLinks(c.arg[Seq[Int]]("ids"))//3
+)
+
+```
+
+</Instruction>
+
+Let's try to understand what is going on in there:
+
+1. As explained previously, we're adding new fields with these names (`link` and `links`)
```

Comment:
This way we have: `allLInks`, `link` and `links` :) 
Maybe this way: `links`, `link-by-id` and `links-by-ids`?

Reply:
it would be refactored later when we will come to resolvers. At the end will be only one function. The reason is to show how it wold be simplified.